python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -0,0 +1,126 @@
+# Gomspace test case. Gets cycle count purely for diagnostic purposes and logs
+# any other Gomspace state fields.
+from .base import SingleSatOnlyCase
+
+
+class GomspaceCheckoutCase(SingleSatOnlyCase):
+
+    def setup_case_singlesat(self, simulation):
+        simulation.flight_controller.write_state(
+            "pan.state", 9)  # Manual state
+        self.run_case_singlesat(simulation)
+        print("Gomspace cases finished.")
+
+    def run_case_singlesat(self, simulation):
+        simulation.cycle_no = simulation.flight_controller.read_state(
+            "pan.cycle_no")
+
+        def read_state(self, string_state):
+            return simulation.flight_controller.read_state(string_state)
+
+        def write_state(self, string_state, state_value):
+            simulation.flight_controller.write_state(string_state, state_value)
+            return read_state(self, string_state)
```

Comment:
Aren't you supposed to use self.read_state(string_state) to call a class method from within a class? idk maybe this works too lol